src/Injector.php

```diff
@@ -42,9 +44,14 @@ class Injector implements InjectorInterface
     /**
      * Constructor
      *
-     * @param null|DefinitionInterface $definition
-     * @param null|InstanceManager $instanceManager
-     * @param null|Config $config
+     * @param ConfigInterface|null $config A custom configuration to utilize. An empty configuration is used
```

Comment:
This is just cosmetic change but I would stick with `null` as first in type list, it's easier to catch it quickly that the value can be nullable.

Reply:
I prefer it the other way around, since I experiences that I'm more interested in the required type most of the time and not its nullability. But this is personal style and I don't mind do adopt a different one.

Although I don't mind changing this (in terms of a general question) - If this is not a must, I'd like to leave it this way for now, since I've got more important thins on my ToDo list.

Reply:
For me the order should be: `null|(simple types in alphabetical order)|(complex types in alphabetical order)` and this is what I did in upcoming zend-coding-standard:
https://github.com/zendframework/zend-coding-standard/blob/feature/sniffs/src/ZendCodingStandard/Helper/Methods.php#L168-L212

Why I think it's right? Because we need to have some order, and as null is not a 'normal type' should be in some fixed position, what we can choose is the first or the last, so I think it's more clear to have it at the beginning, especially when we have couple types. And in case we have multiple nullable params we can see it easily, for example:
```php
/**
 * @param null|DateTime $abc
 * @param null|AnotherType $def
 * @param null|MyLongTypeNameIsHere $ghi
 */
```
For me this looks better than:
```php
/**
 * @param DateTime|null $abc
 * @param AnotherType|null $def
 * @param MyLongTypeNameIsHere|null $ghi
 */
```

Reply:
> …and this is what I did in upcoming zend-coding-standard

At the moment it is only a suggestion. Not accepted! 😉

Reply:
@froschdesign yes, of course, feel free to post your opinion if you have another one 😄